### ISSUE TITLE
code questions

### ISSUE THREAD

**Issue by Like-h** [OrigamiSL/GBT#2]
您好，看了您的论文感觉很受启发，于是打算下载代码跑一下，结果在开始第二阶段训练时，出现了forward() takes 5 positional arguments but 6 were given 的参数不匹配问题。想请问应该如何修改

**Comment by OrigamiSL** [comment]
如果你想使用我的GBT-Vanilla模型，你的attn是不能用Informer的Prob的(注意一下，默认的Prob，我在新的main.py里把默认的改成了Full也就是常规的attn)，因为它的注意力参数没有GBT用的那个对Decoder注意力分数的高斯平滑机制。如果想复现结果，我推荐你使用Script文件夹里面所有的指令跑相应数据集，那些参数都是对的并且和文章里结果是对应上的，你可以看到所有的GBT-Vanilla模型里的注意力都是设为最常规的注意力模型。

**Comment by Like-h** [comment]
Script文件夹里面ETT油温数据的指令seq_len就是输入数据长度吗？seq_len是168然而论文里是Input window size对应的是96。
按照指令python -u main.py --model GBT --data ETTh1 --features M --seq_len 168 --label_len 168 --pred_len 96 --s_layers 3,2,1 --d_layers 2 --attn Full --des 'Exp' --itr 5 --learning_rate 0.0001 --dropout 0.1 --fd_model 16 --d_model 512 --time --instance --criterion Standard --test_inverse运行得到的结果是first stage|Mean|mse:0.4159278907577927, mae:0.42890114442576815|Std|mse:0.007811625893752024, mae:0.004746153163805354
second stage|Mean|mse:0.4206318453165142, mae:0.4342646364472724|Std|mse:0.00949963664472513, mae:0.0067247723899905715
应该是second stage的结果吧？和多变量预测的表格中结果还是有一些不一样。请问我错在哪里，谢谢

**Comment by OrigamiSL** [comment]
您好，我看您也发现了超参设置的问题，所以您应该按照我论文的设置阅读一下main.py对参数的说明并调整一下超参数，scripts里的一些设置确实和论文里不一样，因为个人调试代码的原因（这个我在上一个回复里说错了属实抱歉）。您可以试着根据论文里的超参数重新调整命令来调试程序，比如第一阶段隐藏层维度，输入长度等。为了确保没问题，我刚才又自己跑了一下，肯定是对应的。
我可以给您一个准确的ETTh1多变量预测96的命令，其余的您可以对照着这个和论文相应调整：
python -u main.py --model GBT --data ETTh1 --features M --seq_len 96 --label_len 96 --pred_len 96 --s_layers 3,2,1 --d_layers 2 --attn Full --des 'Exp' --itr 5 --learning_rate 0.0001 --dropout 0.1 --fd_model 32 --d_model 512 --time  --criterion Standard --test_inverse --instance

**Comment by Like-h** [comment]
非常感谢！！



发自我的iPhone


------------------ Original ------------------
From: Origami_SL ***@***.***&gt;
Date: Wed,Sep 20,2023 9:26 PM
To: OrigamiSL/GBT-Neural_Networks_2023 ***@***.***&gt;
Cc: Like-h ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [OrigamiSL/GBT-Neural_Networks_2023] code questions (Issue #2)

**Comment by Like-h** [comment]
我按照您上封邮件中提供的脚本如上运行代码，得到的结果如下，似乎和论文中的MSE：0.398对不上
first stage|Mean|mse:0.42002862272867303, mae:0.43080847272390554|Std|mse:0.005977445002239968, mae:0.003617943314364698
second stage|Mean|mse:0.41749166768150453, mae:0.4308628956903185|Std|mse:0.007244789489096437, mae:0.003920299246132357

**Comment by OrigamiSL** [comment]
2023-09-29-08_06_33|dataset: ETTh1_M|pred_96|stage_first stage|mse:0.4016394298837379, mae:0.420359300389631
2023-09-29-08_06_35|dataset: ETTh1_M|pred_96|stage_second stage|mse:0.39802348245163066, mae:0.4193431058079624
刚跑随便一个itr的结果

**Comment by Like-h** [comment]
可以加个联系方式吗 ？论文里的EMS模块不太清晰。真的很想学习一下您的论文！（沟通不便，后附微信）MXYC_13_32

**Comment by liuqinyong** [comment]
你好，你完整的参数是怎么设置的？我跑的效果差的有点远
first stage|Mean|mse:0.6389647126197815, mae:0.5910306572914124|Std|mse:0.0, mae:0.0
second stage|Mean|mse:0.6295344829559326, mae:0.590165376663208|Std|mse:0.0, mae:0.0
